generators/schema/index.js

```diff
@@ -0,0 +1,239 @@
+  }
+
+  return undefined;
+}
+
+function addRequiredResource(resourceInfo) {
+  if (requiredResources.indexOf(resourceInfo) === -1) {
+    requiredResources.push(resourceInfo);
+  }
+}
+
+module.exports = yeoman.Base.extend({
+  constructor: function () {
+    yeoman.Base.apply(this, arguments);
+
+    this.option('repo-root', {
+      desc: 'The root of the translators repo',
+      alias: 'r',
+      type: String
+    });
+
+    this.props = { devices: [] };
+
+    var resourceChoices = [];
+    var resourceInfos = [];
+    var repoRoot;
+
+    if (this.options['repo-root'] === undefined) {
+      this.log(chalk.bold.red('WARNING: ') + 'No repo specified. Defaulting to ../translators/');
+      repoRoot = '../translators/';
+    } else {
+      repoRoot = this.options['repo-root'];
+    }
+
+    getKnownResources(resourceChoices, resourceInfos, repoRoot);
+
+    this.addResource = function (device, message) {
+      var that = this;
+      var isFirst = device.resources.length === 0;
+      var shouldAdd = function (response) {
+        return isFirst || response.addResource;
+      };
+
+      var prompts = [
+        {
+          when: shouldAdd,
+          type: 'rawlist',
+          name: 'resourceSchema',
+          message: isFirst ? 'What schema does the first resource use?' : 'What schema does the resource use?',
+          choices: resourceChoices
+        },
+        {
+          when: shouldAdd,
+          type: 'input',
+          name: 'resourceName',
+          message: 'What is the resource name (e.g. colorMode)?',
+          validate: utils.validateNotEmpty('Please enter a valid resource name.')
+        },
+        {
+          when: shouldAdd,
+          type: 'input',
+          name: 'displayName',
+          message: 'What is the resource display name (e.g. Color Mode)?',
+          validate: utils.validateNotEmpty('Please enter a valid resource display name.')
+        },
+        {
+          when: shouldAdd,
+          type: 'input',
+          name: 'resourceDescription',
+          message: 'What is the resource description (e.g. Light on/off.)?',
+          validate: utils.validateNotEmpty('Please enter a valid resource description.')
+        },
+        {
+          when: shouldAdd,
+          type: 'confirm',
+          name: 'isWritable',
+          message: 'Is the resource writable'
+        }
+      ];
+
+      if (!isFirst) {
+        prompts.unshift(
+          {
+            type: 'confirm',
+            name: 'addResource',
+            message: message
+          }
+        );
+      }
+
+      return this.prompt(prompts).then(function (props) {
+        if (isFirst || props.addResource) {
+          var resourceInfo = getResourceInfo(props.resourceSchema, resourceInfos);
+          addRequiredResource(resourceInfo);
+          device.resources.push({ schema: resourceInfo, name: props.resourceName, displayName: props.displayName, writable: props.isWritable, description: props.resourceDescription });
```

Comment:
>sources.pus [](start = 19, length = 11)

We allow resources with duplicate names - should we 'smarten' this to catch those errors? I was able to create multiple resources, different schema but same resourceName etc